src/core/sock/sockinfo_ulp.cpp

```diff
@@ -508,6 +533,9 @@ int sockinfo_tcp_ops_tls::setsockopt(int __level, int __optname, const void *__o
             return -1;
         }
     } else {
+#ifdef DEFINED_UTLS
+        xlio_tls_api_setup();
```

Comment:
Current implementation is not thread safe. Such a runtime initialization needs to be protected. consider some kind of thread safe singleton assuming that a pointer assignment is an atomic operation on the target platforms (no need to lock the check for openssl API initialization)